### PR TITLE
"State" field should not be pre-selected

### PR DESCRIPTION
**Pull request by regiskuckaertz**
## Why are you doing this?

Makes the state field mandatory and adds the wiring with the Redux store.

[**Trello Card**](https://trello.com/c/0YPtHKCr/948-npf-state-field-should-not-be-pre-selected)

## Screenshots

screen shot 2018-09-28 at 12 17 26
screen shot 2018-09-28 at 12 17 33

### CODE REVIEW

File: assets/pages/new-contributions-landing/components/ContributionForm.jsx [hunk 1]
```diff
@@ -128,6 +126,7 @@ const isSmallerOrEqual: (number, string) => boolean = (max, input) => parseFloat
 
 const checkFirstName: string => boolean = isNotEmpty;
 const checkLastName: string => boolean = isNotEmpty;
+const checkState: (string | null) => boolean = s => isNotEmpty(s || '');
```
Comment (joelochlann) [inline assets/pages/new-contributions-landing/components/ContributionForm.jsx]: I think I'd find this more readable as
```javascript
typeof s === 'string' && isNotEmpty(s)
```
or
```javascript
s !== null && isNotEmpty(s)
```

File: assets/pages/new-contributions-landing/components/ContributionForm.jsx [hunk 2]
```diff
@@ -75,11 +75,9 @@ type PropTypes = {|
   isDirectDebitPopUpOpen: boolean
 |};
 
-type FormValueType = string | null;
-
 // We only want to use the user state value if the form state value has not been changed since it was initialised,
 // i.e it is null.
-const getCheckoutFormValue = (formValue: FormValueType, userValue: FormValueType): FormValueType =>
+const getCheckoutFormValue = <A>(formValue: A | null, userValue: A | null): A | null =>
```
Comment (joelochlann) [inline assets/pages/new-contributions-landing/components/ContributionForm.jsx]: why was this change necessary when it looks like we're still only passing in `string` to this function?

Comment (regiskuckaertz) [reply assets/pages/new-contributions-landing/components/ContributionForm.jsx]: Good catch, I previously had to check a field of a different type.
